submissions/misshatsker/oop-exercise/index.js

```diff
@@ -0,0 +1,62 @@
+class MyInhabitant {
+   static inhabitantProperties = ['species', 'name', 'gender', 'saying'];
+   static delimiter = '; ';
+
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.properties = [...MyInhabitant.inhabitantProperties];
+   }
+
+   getInhabitantInfo() {
+      const values = this.properties.map((key) => this[key]);
+      return values.join(MyInhabitant.delimiter);
+   }
+}
+
+class Human extends MyInhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, saying);
+      this.hands = 2;
+      this.legs = 2;
+      this.properties.push('hands', 'legs');
+   }
+}
+
+class Animal extends MyInhabitant {
+   constructor(name, gender, saying) {
+      super('animal', name, gender, saying);
+      this.legs = 4;
+      this.properties.push('legs');
```

Comment:
There is no good reason here to change the property the class doesn't own.
Use a method to populate and return a list of class properties.
Use method overloading/overriding to supplement the list provided by a parent class with properties owned by a child class.